src/lang/checkers/Cashflow.ml

```diff
@@ -972,12 +972,12 @@ module ScillaCashflowChecker
                   acc_changes || (get_id_tag arg) <> arg_tag))
               in
           let f_tag = lub_tags (Map res_tag) (Map expected_tag) in
-          let new_f = update_id_tag f f_tag in
-          (Builtin (new_f, final_args),
+          (Builtin (f, final_args),
            res_tag,
            final_field_env,
            final_local_env,
-           changes || f_tag <> get_id_tag f)
+           (*changes || f_tag <> get_id_tag f) -- double check me! *)
+           changes || f_tag <> res_tag)
```

Comment:
`f_tag` and `res_tag` do not represent the same thing, so they should not be checked for inequality.

The problem will be solved once you introduce `builtin_ident`.